app/src/main/java/com/odysee/app/ui/findcontent/FileViewFragment.java

```diff
@@ -1541,87 +1538,59 @@ public void onClick(View view) {
         });
 
         MediaRelativeLayout mediaContainer = root.findViewById(R.id.file_view_media_container);
-        PlayerView playerView = root.findViewById(R.id.file_view_exoplayer_view);
-        TextView playbackSpeed = playerView.findViewById(R.id.player_playback_speed);
-        TextView playbackQuality = playerView.findViewById(R.id.player_quality);
-        playbackSpeed.setText(DEFAULT_PLAYBACK_SPEED);
-        playbackQuality.setText(AUTO_QUALITY_STRING);
+        StyledPlayerView playerView = root.findViewById(R.id.file_view_exoplayer_view);
 
         Context context = getContext();
-        if (context instanceof MainActivity) {
-            MainActivity activity = (MainActivity) context;
-            activity.playerMediaRouteButton = playerView.findViewById(R.id.player_media_route_button);
-            if (activity.mediaRouteButtonVisible) {
-                activity.playerMediaRouteButton.setVisibility(View.VISIBLE);
-            }
-            activity.getCastHelper().setUpCastButton(activity.playerMediaRouteButton);
-        }
-
-        playbackSpeed.setOnCreateContextMenuListener(new View.OnCreateContextMenuListener() {
-            @Override
-            public void onCreateContextMenu(ContextMenu contextMenu, View view, ContextMenu.ContextMenuInfo contextMenuInfo) {
-                Helper.buildPlaybackSpeedMenu(contextMenu);
-            }
-        });
-        playbackSpeed.setOnClickListener(new View.OnClickListener() {
+        playerView.setControllerVisibilityListener(new StyledPlayerView.ControllerVisibilityListener() {
             @Override
-            public void onClick(View view) {
-                Context context = getContext();
-                if (context instanceof MainActivity) {
-                    ((MainActivity) context).openContextMenu(playbackSpeed);
-                }
-            }
-        });
+            public void onVisibilityChanged(int visibility) {
+                if (visibility == View.VISIBLE) {
+                    ((ViewGroup)playbackQuality.getParent()).setVisibility(View.VISIBLE);
+                    MainActivity activity = (MainActivity) context;
+                    if (activity != null) {
+                        activity.playerMediaRouteButton = mediaRouteButton;
+                        if (activity.mediaRouteButtonVisible) {
+                            activity.playerMediaRouteButton.setVisibility(View.VISIBLE);
+                        }
+                        activity.getCastHelper().setUpCastButton(activity.playerMediaRouteButton);
+                    }
```

Comment:
Why is this initialization being done every time the controls are shown?
This should be removed, then this placed above the listener:
```java
        if (context instanceof MainActivity) {
            MainActivity activity = (MainActivity) context;
            activity.playerMediaRouteButton = mediaRouteButton;
            if (activity.mediaRouteButtonVisible) {
                activity.playerMediaRouteButton.setVisibility(View.VISIBLE);
            }
            activity.getCastHelper().setUpCastButton(activity.playerMediaRouteButton);
        }
```

Reply:
You are totally right. Fixed with latest commits. Thank you, Keith